### PR TITLE
Import note from url with markdown

### PR DESCRIPTION
**Pull request by StormBurpee**
Hi guys, 

Thought I'd try and create the feature asked for at #923 
The solution I came up with is quite friendly, and provides an easy way to put in a URL and convert it to Markdown. Let me know what you guys think!

**The new note screen, note the import from url at the bottom**
screen shot 2018-05-26 at 4 59 01 pm

**The new modal, for adding a url and importing**
screen shot 2018-05-26 at 4 59 21 pm

**The result in split markdown view**
screen shot 2018-05-26 at 4 59 34 pm

**The final result of the imported note**
screen shot 2018-05-26 at 4 59 51 pm

### CODE REVIEW

File: browser/main/modals/CreateMarkdownFromURLModal.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+  validateUrl(str) {
+    if(/^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})).?)(?::\d{2,5})?(?:[/?#]\S*)?$/i.test(str)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  confirm () {
```
Comment (sosukesuzuki) [inline browser/main/modals/CreateMarkdownFromURLModal.js]: please add a test!

**Comment by StormBurpee** [comment]
Moved the function to the dataApi folder, so that a test can be created easier. But I'm struggling to write the test at the moment, so I'll take a look a little later. If someone want's to take a look, feel free too.

File: browser/main/lib/dataApi/createNoteFromUrl.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+const http = require('http')
+const https = require('https')
+const TurndownService = require('turndown')
+const createNote = require('./createNote')
+
+import { hashHistory } from 'react-router'
+import ee from 'browser/main/lib/eventEmitter'
+
+function validateUrl(str) {
```
Comment (ehhc) [inline browser/main/lib/dataApi/createNoteFromUrl.js]: This method could be simplyfied to 
```
return str.matches(new RegExp('^(?:.......'));
```

Comment (ehhc) [reply browser/main/lib/dataApi/createNoteFromUrl.js]: Oh, my idea would have introduced the opportunity for null pointers.. So it's good that you ignored my concern :D

**Comment by sosukesuzuki** [comment]
@StormBurpee 
please follow eslint(`tests/dataApi/createNoteFromUrl-test.js`).
please fix conflict.

(I'll review the test written by you, later.)

**Comment by Rokt33r** [comment]
Anyway, it looks great first step of the feature. Nice work!

**Comment by sosukesuzuki** [comment]
@StormBurpee please confirm our reviews!

**Comment by Rokt33r** [comment]
I'm going to close this issue because of the inactivity.

**Pull request closed without merging**